yt_dlp/extractor/niconico.py

```diff
@@ -328,36 +344,41 @@ def ping():
 
         return info_dict, heartbeat_info_dict
 
-    def _extract_format_for_quality(self, api_data, video_id, audio_quality, video_quality):
-        def parse_format_id(id_code):
-            mobj = re.match(r'''(?x)
-                    (?:archive_)?
-                    (?:(?P<codec>[^_]+)_)?
-                    (?:(?P<br>[\d]+)kbps_)?
-                    (?:(?P<res>[\d+]+)p_)?
-                ''', '%s_' % id_code)
-            return mobj.groupdict() if mobj else {}
-
-        protocol = 'niconico_dmc'
-        format_id = '-'.join(map(lambda s: remove_start(s['id'], 'archive_'), [video_quality, audio_quality]))
-        vdict = parse_format_id(video_quality['id'])
-        adict = parse_format_id(audio_quality['id'])
-        resolution = try_get(video_quality, lambda x: x['metadata']['resolution'], dict) or {'height': vdict.get('res')}
-        vbr = try_get(video_quality, lambda x: x['metadata']['bitrate'], float)
+    def _extract_format_for_quality(self, video_id, audio_quality, video_quality, dmc_protocol):
+
+        if not audio_quality['isAvailable'] or not video_quality['isAvailable']:
+            return None
+
+        def extract_video_quality(video_quality):
+            # Example: 480p | 0.9M
+            r = re.match(r'^.*\| ([0-9]*\.?[0-9]*[MK])', video_quality)
```

Comment:
This is the same as:
```suggestion
            r = re.search(r'\| ([0-9]*\.?[0-9]*[MK])', video_quality)
```